src/backend/marsha/core/api/timed_text_track.py

```diff
@@ -32,40 +51,52 @@ def get_permissions(self):
             permission_classes = [
                 permissions.IsTokenInstructor
                 | permissions.IsTokenAdmin
+                | permissions.IsParamsVideoAdminOrInstructorThroughPlaylist
                 | permissions.IsParamsVideoAdminThroughOrganization
-                | permissions.IsParamsVideoAdminThroughPlaylist
             ]
-        else:
+        elif self.action in [
+            "destroy",
+            "initiate_upload",
+            "retrieve",
+            "update",
+            "partial_update",
+        ]:
             permission_classes = [
                 permissions.IsTokenResourceRouteObjectRelatedVideo
-                & permissions.IsTokenInstructor
-                | permissions.IsTokenResourceRouteObjectRelatedVideo
-                & permissions.IsTokenAdmin
-                | permissions.IsRelatedVideoPlaylistAdmin
+                & (permissions.IsTokenInstructor | permissions.IsTokenAdmin)
+                | permissions.IsRelatedVideoPlaylistAdminOrInstructor
                 | permissions.IsRelatedVideoOrganizationAdmin
             ]
+        elif self.action is None:
+            if self.request.method not in self.allowed_methods:
+                raise MethodNotAllowed(self.request.method)
+            permission_classes = self.permission_classes
```

Comment:
FMI, how can self.action be none ?

Reply:
`self.action` is `None` at least when you try to make an action without the proper method. I did not investigate it more, but for instance, calling `initiate_upload` action with a PUT method results in a `self.action is None` :shrug: May be it worth some time to dig it.